only split on first namespace =

anneeb opened the pull request.

anneeb committed:
only split on first namespace =

anneeb committed:
make it a function

brianru commented:
Would this work with the example in the bug report? In that example the last `=` is the end of the namespace. https://github.com/stardog-union/stardog-studio/issues/1245

anneeb commented:
@brianru it should

```
> const str = '1=2=3=4=5='
undefined
> const [a,...b]=str.split('=')
undefined
> b
[ '2', '3', '4', '5', '' ]
> b.join('=');
'2=3=4=5='
```

EDIT:  it also works for multiple trails
```
> const str2 = '1=2=3=4=5===='
undefined
> const [c,...d]=str2.split('=')
undefined
> d
[ '2', '3', '4', '5', '', '', '', '' ]
> d.join('=')
'2=3=4=5===='
```

anneeb committed:
add tests for splitNamespace

brianru commented on packages/stardog-language-utils/src/__tests__/namespaceUtils.test.ts:
```diff
@@ -1,10 +1,26 @@
 
+describe('splitNamespace', () => {
+  it('splits a namespace into a string array of the alias and prefix', () => {
+    const namespace = 'foo=http://foo.org/';
```
heh, thanks. i initially forgot these look like `<abbreviation>=<namespace iri>`

brianru commented on packages/stardog-language-utils/src/namespaceUtils.ts:
```diff
@@ -21,6 +21,11 @@ const abbreviate = (
 };
 
+export const splitNamespace = (namespace: string) => {
+  const [alias, ...splitPrefix] = namespace.split('=');
```
simpler to use the `limit` 2nd param to `split`?

anneeb replied:
limit does not return the leftover text

brianru replied:
weird

anneeb replied:
```
'1=2=3='.split('=',2)
>[ '1', '2' ]
```

brianru reviewed (commented):
🐒

brianru reviewed (approved):
🐒
